src/processes.c

```diff
@@ -25,2153 +25,3026 @@
+  regex_t *re;
+#endif
 
-	struct procstat   *next;
-	struct procstat_entry_s *instances;
+  /* Linux Delay Accounting. Unit is ns/s. */
+  gauge_t delay_cpu;
+  gauge_t delay_blkio;
+  gauge_t delay_swapin;
+  gauge_t delay_freepages;
+
+  bool report_fd_num;
+  bool report_maps_num;
+  bool report_ctx_switch;
+  bool report_delay;
+
+  procstat_gauges_t gauges;
+  procstat_counters_t counters;
+  struct procstat *next;
+  struct procstat_entry_s *instances;
 } procstat_t;
 
-static procstat_t *list_head_g = NULL;
+static procstat_t *list_head_g;
+static _Bool report_ctx_switch = 0;
+typedef struct
+{
+    _Bool ps_count;
+    _Bool ps_vm;
+    _Bool ps_rss;
+    _Bool ps_data;
+    _Bool ps_code;
+    _Bool ps_stacksize;
+    _Bool ps_cputime;
+    _Bool ps_pagefaults;
+    _Bool ps_disk_octets;
+    _Bool ps_disk_ops;
+    _Bool cswitch_vol;
+    _Bool cswitch_invol;
+} want_detail_configuration_t;
+static want_detail_configuration_t want_detail_configuration_g;
+static _Bool some_detail_active_g = 0;
+static bool want_init = true;
+static bool report_ctx_switch;
+static bool report_fd_num;
+static bool report_maps_num;
+static bool report_delay;
 
 #if HAVE_THREAD_INFO
 static mach_port_t port_host_self;
 static mach_port_t port_task_self;
 
 static processor_set_name_array_t pset_list;
-static mach_msg_type_number_t     pset_list_len;
+static mach_msg_type_number_t pset_list_len;
 /* #endif HAVE_THREAD_INFO */
 
 #elif KERNEL_LINUX
 static long pagesize_g;
+static void ps_fill_details(const procstat_t *ps, procstat_entry_t *entry);
 /* #endif KERNEL_LINUX */
 
-#elif HAVE_LIBKVM_GETPROCS && HAVE_STRUCT_KINFO_PROC_FREEBSD
+#elif HAVE_LIBKVM_GETPROCS &&                                                  \
+    (HAVE_STRUCT_KINFO_PROC_FREEBSD || HAVE_STRUCT_KINFO_PROC_OPENBSD)
 static int pagesize;
-/* #endif HAVE_LIBKVM_GETPROCS && HAVE_STRUCT_KINFO_PROC_FREEBSD */
+/* #endif HAVE_LIBKVM_GETPROCS && (HAVE_STRUCT_KINFO_PROC_FREEBSD ||
+ * HAVE_STRUCT_KINFO_PROC_OPENBSD) */
 
 #elif HAVE_PROCINFO_H
-static  struct procentry64 procentry[MAXPROCENTRY];
-static  struct thrdentry64 thrdentry[MAXTHRDENTRY];
+static struct procentry64 procentry[MAXPROCENTRY];
+static struct thrdentry64 thrdentry[MAXTHRDENTRY];
 static int pagesize;
 
 #ifndef _AIXVERSION_610
-int     getprocs64 (void *procsinfo, int sizproc, void *fdsinfo, int sizfd, pid_t *index, int count);
-int     getthrds64( pid_t, void *, int, tid64_t *, int );
+int getprocs64(void *procsinfo, int sizproc, void *fdsinfo, int sizfd,
+               pid_t *index, int count);
+int getthrds64(pid_t, void *, int, tid64_t *, int);
 #endif
-int getargs (struct procentry64 *processBuffer, int bufferLen, char *argsBuffer, int argsLen);
+int getargs(void *processBuffer, int bufferLen, char *argsBuffer, int argsLen);
 #endif /* HAVE_PROCINFO_H */
 
+#if HAVE_LIBTASKSTATS
+static ts_t *taskstats_handle;
+#endif
+
+static derive_t ps_delta(derive_t value) {
+	return (value == -1) ? 0 : value;
```

Comment:
`-1` is used as a magic value throughout the code. Please define a constant (VALUE_UNSET?) and use it whenever these variables are initialized, set, or compared to -1 like here.